<a href="https://colab.research.google.com/github/hannamakarova/Andvanced-Analytics-Prediction-Models/blob/main/Dummy_Variables_Credit_Scoring.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

We are working on a credit scoring model for a bank. Information about customers:

Employment_Type — the type of employment of the customer (categories Full-time, Part-time, and Unemployed);
Marital_Status — the marital status of the customer (categories Single, Married, and Divorced).
The target variable is the customer’s credit score (Credit_Score).

Create a linear regression model to predict the credit score based on two categorical variables, having previously converted them to numeric values. Derive the coefficients of this model and provide their interpretation.

In [4]:
import pandas as pd

data = pd.DataFrame({
    "Client_ID": [1, 2, 3, 4, 5, 6, 7, 8, 9, 10],
    "Employment_Type": ["Full-time", "Part-time", "Unemployed", "Full-time", "Part-time",
                        "Unemployed", "Full-time", "Part-time", "Unemployed", "Full-time"],
    "Marital_Status": ["Single", "Married", "Divorced", "Single", "Married",
                       "Single", "Divorced", "Married", "Single", "Divorced"],
    "Credit_Score": [750, 680, 600, 720, 710, 580, 770, 690, 620, 740]

})

data = pd.get_dummies(data, columns=["Employment_Type", "Marital_Status"],drop_first= True)
print(data)


   Client_ID  Credit_Score  Employment_Type_Part-time  \
0          1           750                      False   
1          2           680                       True   
2          3           600                      False   
3          4           720                      False   
4          5           710                       True   
5          6           580                      False   
6          7           770                      False   
7          8           690                       True   
8          9           620                      False   
9         10           740                      False   

   Employment_Type_Unemployed  Marital_Status_Married  Marital_Status_Single  
0                       False                   False                   True  
1                       False                    True                  False  
2                        True                   False                  False  
3                       False                   False   

In [7]:
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LogisticRegression

# prepare data
X = data.drop(["Client_ID","Credit_Score"], axis=1)
y = data["Credit_Score"]

# split data into train and test
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.3, random_state=2024)

# train model
model = LogisticRegression()
model.fit(X_train, y_train)

# make predictions
y_pred = model.predict(X_test)

# get coefficients
coefficients = pd.DataFrame({
    "Feature": X.columns,
    "Coefficient": model.coef_[0]
}).sort_values(by="Coefficient", ascending=False)

print("\nFeature Importance:")
print(coefficients)



Feature Importance:
                      Feature  Coefficient
1  Employment_Type_Unemployed     0.696271
3       Marital_Status_Single     0.376050
0   Employment_Type_Part-time    -0.150901
2      Marital_Status_Married    -0.150901


From the results, we can see that:

* Unemployed individuals are predicted to have a higher credit score than full-time employed individuals, which is not logically consistent with typical financial assumptions. This could be due to data imbalance, missing variables, or model bias.

* Part-time workers have a 15% lower chance (based on a coefficient of –0.151) of receiving a higher credit score compared to full-time workers, suggesting employment type influences credit evaluation.

* Single individuals are associated with a higher credit score (+0.376) compared to divorced individuals, while married individuals show a slight decrease (–0.151) in credit score, also relative to divorced individuals.

